frontend/model/contracts/group.js

```diff
@@ -262,30 +273,39 @@ DefineContract({
       }
     },
     'gi.contracts/group/proposal': {
-      validate: objectOf({
-        proposalType: proposalType,
-        proposalData: object, // data for Vue widgets
-        votingRule: ruleType,
-        expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
-      }),
-      process ({ data, meta, hash }, { state }) {
-        // make sure this isn't a duplicate proposal
+      validate: (data, { state, meta }) => {
+        objectOf({
+          proposalType: proposalType,
+          proposalData: object, // data for Vue widgets
+          votingRule: ruleType,
+          expires_date_ms: number // calculate by grabbing proposal expiry from group properties and add to `meta.createdDate`
+        })(data)
+
+        // Validate this isn't a duplicate proposal
         for (const hash in state.proposals) {
           const prop = state.proposals[hash]
-          if (prop.status === STATUS_OPEN &&
-            prop.data.proposalType === data.proposalType &&
-            prop.meta.username === meta.username &&
-            deepEqualJSONType(prop.data.proposalData, data.proposalData)
+          if (prop.status !== STATUS_OPEN || prop.data.proposalType !== data.proposalType) {
+            continue
+          }
+
+          if (prop.data.proposalType === PROPOSAL_REMOVE_MEMBER &&
+            prop.data.proposalData.member === data.member
           ) {
-            throw new TypeError(`proposal: is identical to proposal ${hash}`)
+            throw new TypeError(L('There is an open proposal to remove them.'))
+          }
+
+          if (deepEqualJSONType(prop.data.proposalData, data.proposalData)) {
```

Comment:
Just a reminder to remove the proposal reason from the comparison (you can use the new `_.omit` func in giLodash)